services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/UserGroupController.java

```diff
@@ -0,0 +1,265 @@
+import com.sourcefuse.jarc.core.constants.CommonConstants;
+import com.sourcefuse.jarc.core.dto.Count;
+import com.sourcefuse.jarc.services.usertenantservice.auth.IAuthUserWithPermissions;
+import com.sourcefuse.jarc.services.usertenantservice.dto.Group;
+import com.sourcefuse.jarc.services.usertenantservice.dto.UserGroup;
+import com.sourcefuse.jarc.services.usertenantservice.enums.RoleKey;
+import com.sourcefuse.jarc.services.usertenantservice.repository.GroupRepository;
+import com.sourcefuse.jarc.services.usertenantservice.repository.UserGroupsRepository;
+import jakarta.validation.Valid;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+@RestController
+@Slf4j
+@RequestMapping("/groups")
+@RequiredArgsConstructor
+public class UserGroupController {
+
+  private final GroupRepository groupRepository;
+
+  private final UserGroupsRepository userGroupsRepo;
+
+  @Value("${one.owner.msg}")
+  private String oneOwnerMsg;
+
+  @Value("${user.group.not.found}")
+  private String usrGrpNotFound;
+
+  @Value("${only.grp.owner.access}")
+  private String grpOwnerAccess;
+
+  @PostMapping("{id}/user-groups")
+  public ResponseEntity<Object> createRole(
+    @Valid @RequestBody UserGroup userGroup,
+    @PathVariable("id") UUID id
+  ) {
+    Optional<UserGroup> usrGr;
+    Group svdGroup;
+    Optional<Group> group = groupRepository.findById(id);
+    if (group.isPresent()) {
+      svdGroup = group.get();
+      if (userGroup.getGroup().getId() == null) {
+        userGroup.getGroup().setId(svdGroup.getId());
+      }
+      usrGr =
+        userGroupsRepo.findByGroupIdAndUserTenantId(
+          userGroup.getGroup().getId(),
+          userGroup.getUserTenant().getId()
+        );
+      if (!usrGr.isPresent()) {
+        usrGr = (Optional.ofNullable(userGroupsRepo.save(userGroup)));
+        group.get().setModifiedOn(usrGr.get().getModifiedOn());
+        groupRepository.save(group.get());
```

Comment:
does this required, since we are not updating the group directly we are creating the user group and its updating the relation of group